### PR TITLE
gh-93223: More aggressive Jump-To-Jump elimination

### PR DESCRIPTION
**Pull request by sweeneyde**
https://github.com/python/cpython/issues/93223

### CODE REVIEW

File: Lib/test/test_dis.py [hunk 1]
```diff
@@ -1392,100 +1392,99 @@ def _prepare_test_cases():
...
   Instruction(opname='LOAD_CONST', opcode=100, arg=3, argval=6, argrepr='6', offset=74, starts_line=None, is_jump_target=False, positions=None),
   Instruction(opname='COMPARE_OP', opcode=107, arg=4, argval='>', argrepr='>', offset=76, starts_line=None, is_jump_target=False, positions=None),
-  Instruction(opname='POP_JUMP_FORWARD_IF_FALSE', opcode=114, arg=2, argval=88, argrepr='to 88', offset=82, starts_line=None, is_jump_target=False, positions=None),
+  Instruction(opname='POP_JUMP_BACKWARD_IF_FALSE', opcode=175, arg=28, argval=28, argrepr='to 28', offset=82, starts_line=None, is_jump_target=False, positions=None),
   Instruction(opname='POP_TOP', opcode=1, arg=None, argval=None, argrepr='', offset=84, starts_line=8, is_jump_target=False, positions=None),
-  Instruction(opname='JUMP_FORWARD', opcode=110, arg=14, argval=116, argrepr='to 116', offset=86, starts_line=None, is_jump_target=False, positions=None),
-  Instruction(opname='JUMP_BACKWARD', opcode=140, arg=31, argval=28, argrepr='to 28', offset=88, starts_line=7, is_jump_target=True, positions=None),
```
Comment (sweeneyde) [inline Lib/test/test_dis.py:1398]: This instruction gets deleted altogether

File: Lib/test/test_dis.py [hunk 2]
```diff
@@ -1392,100 +1392,99 @@ def _prepare_test_cases():
   Instruction(opname='LOAD_FAST', opcode=124, arg=0, argval='i', argrepr='i', offset=72, starts_line=7, is_jump_target=True, positions=None),
   Instruction(opname='LOAD_CONST', opcode=100, arg=3, argval=6, argrepr='6', offset=74, starts_line=None, is_jump_target=False, positions=None),
   Instruction(opname='COMPARE_OP', opcode=107, arg=4, argval='>', argrepr='>', offset=76, starts_line=None, is_jump_target=False, positions=None),
-  Instruction(opname='POP_JUMP_FORWARD_IF_FALSE', opcode=114, arg=2, argval=88, argrepr='to 88', offset=82, starts_line=None, is_jump_target=False, positions=None),
```
Comment (sweeneyde) [inline Lib/test/test_dis.py:1395]: This jump-to-jump gets folded together

File: Lib/test/test_dis.py [hunk 3]
```diff
@@ -1392,100 +1392,99 @@ def _prepare_test_cases():
...
+  Instruction(opname='LOAD_CONST', opcode=100, arg=0, argval=None, argrepr='None', offset=256, starts_line=25, is_jump_target=False, positions=None),
+  Instruction(opname='LOAD_CONST', opcode=100, arg=0, argval=None, argrepr='None', offset=258, starts_line=None, is_jump_target=False, positions=None),
   Instruction(opname='LOAD_CONST', opcode=100, arg=0, argval=None, argrepr='None', offset=260, starts_line=None, is_jump_target=False, positions=None),
-  Instruction(opname='LOAD_CONST', opcode=100, arg=0, argval=None, argrepr='None', offset=262, starts_line=None, is_jump_target=False, positions=None),
-  Instruction(opname='CALL', opcode=171, arg=2, argval=2, argrepr='', offset=264, starts_line=None, is_jump_target=False, positions=None),
-  Instruction(opname='POP_TOP', opcode=1, arg=None, argval=None, argrepr='', offset=274, starts_line=None, is_jump_target=False, positions=None),
-  Instruction(opname='JUMP_FORWARD', opcode=110, arg=11, argval=300, argrepr='to 300', offset=276, starts_line=None, is_jump_target=False, positions=None),
```
Comment (sweeneyde) [inline Lib/test/test_dis.py:1446]: A jump-to-jump gets eliminated here

**Comment by markshannon** [comment]
My only concern with this PR is that I would like to remove all backwards conditional jumps: https://github.com/faster-cpython/ideas/discussions/297#discussioncomment-2626297.
Having said that, we'll need to handle conditional backward jumps anyway, so this seems like an improvement even then.

**Comment by markshannon** [comment]
My only concern with this PR is that I would like to remove all backwards conditional jumps: https://github.com/faster-cpython/ideas/discussions/297#discussioncomment-2626297.
Having said that, we'll need to handle conditional backward jumps anyway, so this seems like an improvement even then.

**Pull request merged**